codalab/lib/bundle_cli.py

```diff
@@ -600,20 +600,33 @@ def resolve_key_targets(self, client, worksheet_uuid, target_specs):
         Helper: target_specs is a list of strings which are [<key>]:<target>
         Returns: [(key, worker.download_util.BundleTarget), ...]
         """
-        keys = set()
+
+        def is_ancestor(path1, path2):
```

Comment:
`is_ancestor_or_descendant` is probably clearer.  Note that if `path1 == path2`, we return false.  Is that the right behavior?

Reply:
This should be okay because we check for duplicate keys before this check.